crates/biome_js_analyze/src/semantic_analyzers/nursery/use_sorted_classes.rs

```diff
@@ -0,0 +1,1373 @@
+                    {
+                        let attributes_option: Vec<String> = attributes_option; // TODO: is there a better way to do this?
+                        result.attributes.extend(attributes_option);
+                    }
+                }
+                "functions" => {
+                    if let Some(functions) =
+                        Deserializable::deserialize(&value, &key_text, diagnostics)
+                    {
+                        result.functions = functions;
+                    }
+                }
+                "preset" => {
+                    if let Some(preset_option) =
+                        Deserializable::deserialize(&value, &key_text, diagnostics)
+                    {
+                        let preset_option: String = preset_option; // TODO: is there a better way to do this?
+                        let preset_option = preset_option.as_str();
+                        match preset_option {
+                            "tailwind-css" => {
+                                preset = UseSortedClassesPreset::TailwindCSS;
+                            }
+                            "no-preset" => {
+                                preset = UseSortedClassesPreset::None;
+                            }
+                            _ => {
+                                diagnostics.push(DeserializationDiagnostic::new_unknown_value(
+                                    preset_option,
+                                    value.range(),
+                                    ALLOWED_PRESETS,
+                                ));
+                            }
+                        }
+                    }
+                }
+                "utilities" => {
+                    if let Some(utilities_opt) =
+                        Deserializable::deserialize(&value, &key_text, diagnostics)
+                    {
+                        utilities_option = Some(utilities_opt);
+                    }
+                }
+                unknown_key => diagnostics.push(DeserializationDiagnostic::new_unknown_key(
+                    unknown_key,
+                    key.range(),
+                    ALLOWED_OPTIONS,
+                )),
+            }
+        }
+
+        let resolved_utilities = match utilities_option {
+            Some(utilities) => utilities,
+            None => get_utilities_preset(&preset),
+        };
+        result.utilities = resolved_utilities
+            .iter()
+            .flat_map(|layer| {
+                // TODO: extend layer here
+                layer.classes.clone()
+            })
+            .collect();
+
+        Some(result)
+    }
+}
+
+// presets
+// -------
+
+#[derive(Debug, Default, Clone)]
+pub enum UseSortedClassesPreset {
+    None,
+    #[default]
+    TailwindCSS,
+}
+
+const ALLOWED_PRESETS: &[&str] = &["no-preset", "tailwind-css"];
+
+// TODO: move to separate file?
+fn get_utilities_preset(preset: &UseSortedClassesPreset) -> Vec<UtilityLayer> {
```

Comment:
Yes, I was worried about this too, even asked around in the Discord server. Thanks for the help!

It was annoying to me that I couldn't figure out how to prevent those allocations lol.

Reply:
I believe this was solved. We can probably avoid more string allocations in a future pass. I made a note in the issue where I keep a centralized TODO list. Resolving this.